### PR TITLE
Finish and merge in @Jngai Notes + Help Bubble ES6 PR

### PR DESCRIPTION
**Pull request by alexsoble**
# Who is this PR for?

Developers who want to contribute to the codebase by adding or improving client-side features. 

# What problem does this PR fix?

Not all our JS files use ES6 modules, which can be confusing for new JS devs coming on to the project.

# What does this PR do?

Merges in #1311 and adds some changes on top of it. 

Updates the following files to use ES6 modules and follow create-react-app naming conventions:

+ RestrictedNotesPageContainer.js
+ NotesDetails.js
+ Api.js
+ HelpBubble.js

Also adds initial test coverage for:

+ Api.js
+ HelpBubble.js
# Merge Checklist

## Javascript QA

+ [x] Checked latest in IE - Create/Edit Notes
+ [x] Checked latest in IE - Create/Edit Restricted Notes
+ [x] Checked latest in IE - Help Bubble on Academic Detail

## Specs 

+ [x] Included specs for new code.

### CODE REVIEW

**Comment by Jngai** [comment]
Hey @alexsoble I notice that my contributions are not showing up on student insight's contributions graph. If you look at my commits closely I have no pic attached. I got a new computer a couple of months ago so my git email may not have been set up properly to get the credit for my commits. Is there anything you can do as the maintainer of the project?

**Comment by alexsoble** [comment]
@Jngai Is that because your git email didn't match your github email?

**Comment by alexsoble** [comment]
@Jngai Yeah, here's what the commit UI says:

screen shot 2017-12-18 at 10 36 58 am

And here's the help article they point to:

https://help.github.com/articles/setting-your-commit-email-address-in-git/

**Comment by Jngai** [comment]
@alexsoble I tried that already and my email is pointing to the correct one. See here

screen shot 2017-12-18 at 11 39 27 am
screen shot 2017-12-18 at 11 39 03 am

**Comment by alexsoble** [comment]
Try `git log`, when I compare my commits and yours I see no quotes around my email address but quotes around yours:

```

commit 5b6241217a6cc94e0fb4771ffdd60cc2d505e075
Author: Alex Soble <asoble@gmail.com>
Date:   Mon Dec 18 10:02:21 2017 -0600

    Remove RestrictedNotesPageContainer from legacy.js

commit 873e1fa2cc1f317082619ad4069fa325594c94d3
Author: Judy Ngai <“judy.ngai1228@gmail.com”>
Date:   Wed Dec 13 11:23:40 2017 -0500

    fix indentation issue
```

**Comment by alexsoble** [comment]
@Jngai Happy to work on this with you but we should move this topic out of this PR, discussion on this PR should be mostly about the ES6 refactoring. 👍

**Comment by Jngai** [comment]
@alexsoble I think I fixed it. Its a typo mistake on `git config --global user.email`

**Comment by alexsoble** [comment]
FYI was having some issues with my local git and remote GitHub seemingly out of sync when it comes to what files are capitalized, fixed that with running this update locally:

```
git config core.ignorecase false
```

That fixed the issue for me. Might be useful for other folks who contribute to the camelcasing part of the create-react-app-style refactor work.

**Comment by alexsoble** [comment]
Of the four refactored components:

+ NotesDetails and RestrictedNotesPageContainer have JS specs
+ Api and HelpBubble do not have JS specs

Since we decided to make "adds new tests for untested code" a checklist item for PRs in #1289, I should add specs to Api.js and HelpBubble.js before reviewing this PR.

**Comment by alexsoble** [comment]
Alright — create-react-app naming conventions applied, legacy.js updated, test coverage added when missing. 

Next up I'm going to spin up an VM and manually test the key functionality in IE:

+ [ ] Checked latest in IE - Create/Delete/Edit Notes
+ [ ] Checked latest in IE - Create/Delete/Edit Restricted Notes
+ [ ] Checked latest in IE - Help Bubble on Academic Detail

**Comment by alexsoble** [comment]
Ooooh, right off the bat it looks like there's an issue with the HelpBubble not opening.

**Comment by alexsoble** [comment]
Here's what I get when I try to open a HelpBubble:

screen shot 2017-12-18 at 2 02 12 pm

**Comment by alexsoble** [comment]
It's the same function-binding issue we've had on past PRs in this sequence of work.

**Comment by alexsoble** [comment]
Alright, re-running the QA now!

**Comment by alexsoble** [comment]
# Help Bubble Working Now (IE)
help-bubble-working-ie

# Regular Note, Editing (IE)
regular-editing-ie

# Regular Note, Creating (IE)
regular-creating-ie

# Restricted Note, Editing (IE)
restricted-edited-ie

# Regular Note, Creating (IE)
restricted-created-ie

**Comment by alexsoble** [comment]
Alright, second QA passed. Time to merge, deploy, and verify.

**Comment by alexsoble** [comment]
I logged into production in IE and opened/closed a few Help Bubbles, logged in to the demo Heroku site in IE and created/edited a few regular/restricted notes.

File: app/assets/javascripts/student_profile/Api.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+class Api extends React.Component {
```
Comment (kevinrobinson) [inline app/assets/javascripts/student_profile/Api.js]: This shouldn't be a React component, it's not rendering UI to the DOM.

To translate what this was before, we can write each function like this:

```
export function saveNotes(...) { ... }
export function deleteEventNoteAttachment(...) { ... }
```

And then underscore-prefixed methods then don't need to be exported at all (or if you want to for testing purposes, you can export them as underscore still).

**Comment by alexsoble** [comment]
@kevinrobinson How about leaving it as a class and dropping the extends
React.Component bit?

File: app/assets/javascripts/student_profile/Api.js (hunk 1 above)
Comment (kevinrobinson) [reply app/assets/javascripts/student_profile/Api.js]: Ah I read this wrong and thought Api was static.  This is great here, just don't need `extends React.Component` and the props bits.

**Comment by kevinrobinson** [comment]
@alexsoble 👍 yeah I read this wrong, thanks for helping me see that!

**Pull request merged**